ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -131,6 +133,9 @@ public int compare(IndexEntry e1, IndexEntry e2) {
   // When undelete is enabled, DELETE is not the final state of the blob, since a DELETEd blob can be UNDELTEd.
   private final boolean isDeleteFinalStateOfBlob;
 
+  private final AtomicInteger partialLogSegmentCount = new AtomicInteger(0);
```

Comment:
nit: Does this need to be AtomicInteger? I guess `BlobStoreCompactor` call `updatePartialLogSegmentInfo` only one partition at a time.

Reply:
It's about the thread which reads the variable. The ingraph metric collector reads it while the compactor is updating it.
We can argue that JVM make at least 32bit primary type atomic. But it's good practice to use atomic since different threads are reading and updating it.